### PR TITLE
Configure default http client timeout

### PR DESCRIPTION
**Pull request by pankrator**
## Motivation

There are a few different aspects of these timeouts:

1. OSB calls via the /v1/osb API of Service Manager:
  - Calls for get catalog are done via CatalogFetcher which uses the default http client which was not configured.
  - Calls for all other requests are done via reverse proxy, which does not know when it should cancel the request, because the context is not cancelled on server.request_timeout. The only time it will time out is when the headers are not yet received. But if there is a "bad broker" which has returned headers, but is streaming the response very slowly, we will wait for the response forever. This can be fixed with something like [http.TimeoutHandler](https://golang.org/pkg/net/http/#TimeoutHandler). There is a branch with implementation: https://github.com/Peripli/service-manager/tree/handler_timeouts.
2. OSB calls via SM Write APIs
  - Calls are done via osb client which is initialized with http client that has configured timeout for the whole request, so they do not suffer the above problem

If you need to reproduce some of the scenarios let me know and I will attach the broker servers and scripts to reproduce.

## Pull Request status

* [x] Initial implementation
* [ ] Refactoring
* [x] Unit tests

### CODE REVIEW

File: test/common/application.yml [hunk 1]
```diff
@@ -3,10 +3,11 @@ server:
   shutdown_timeout: 4000ms
   port: 1234
 httpclient:
-  response_header_timeout: 40000ms
-  tls_handshake_timeout: 4000ms
-  idle_conn_timeout: 4000ms
-  dial_timeout: 4000ms
+  timeout: 1000ms
```
Comment (NickyMateev) [inline test/common/application.yml]: Aren't these values too low?

File: test/broker_test/broker_test.go [hunk 2]
```diff
@@ -571,6 +572,25 @@ var _ = test.DescribeTestsFor(test.TestCase{
 						Expect(transitiveResourcesActualCount).To(Equal(transitiveResourcesExpectedCount))
 					})
 				})
+
+				Context("when broker is responding slow", func() {
+					It("should timeout", func() {
+						brokerServer.CatalogHandler = func(rw http.ResponseWriter, req *http.Request) {
+							rw.WriteHeader(http.StatusOK)
+							if fl, ok := rw.(http.Flusher); ok {
+								for i := 0; i < 30; i++ {
```
Comment (NickyMateev) [inline test/broker_test/broker_test.go]: I see we get very innovative with our testing as time goes on :D

File: test/common/application.yml (hunk 1 above)
Comment (pankrator) [reply test/common/application.yml]: mm may be they will be low for Travis tests... I will try to adjust them if they fail.

File: test/broker_test/broker_test.go (hunk 2 above)
Comment (pankrator) [reply test/broker_test/broker_test.go]: Will this one get in unnoticed :D

Comment (KirilKabakchiev) [reply test/broker_test/broker_test.go]: noticed ! :)

File: pkg/httpclient/settings.go [hunk 3]
```diff
@@ -64,6 +69,7 @@ func (s *Settings) Validate() error {
 func Configure(settings *Settings) {
 	transport := http.DefaultTransport.(*http.Transport)
 
+	http.DefaultClient.Timeout = settings.Timeout
```
Comment (KirilKabakchiev) [inline pkg/httpclient/settings.go]: we use the default transport in some places in the code (without the default client).  Have you checked if what we configured on it is sufficient ?

**Pull request merged**